src/DropDownMenu/DropDownMenu.js

```diff
@@ -223,23 +225,62 @@ class DropDownMenu extends Component {
   };
 
   handleRequestCloseMenu = () => {
-    this.setState({
-      open: false,
-      anchorEl: null,
-    });
+    this.close(false);
+  };
+
+  handleFocus = (event) => {
+    // this is a work-around for SelectField losing keyboard focus
+    // because the containing TextField re-renders
+    if (event) event.stopPropagation();
```

Comment:
@petermikitsh this is the reason that TextField doesn't also show its highlight animation. When the textbox receives a focus or blur event it does a `setState` which re-renders, and the drop down was losing focus when that happened. I tried for a while to get it to work that way but couldn't. Do you want to take a look?